lib/modules/auth/login/bloc/login_repository.dart

```diff
@@ -0,0 +1,46 @@
+import 'package:dio/dio.dart';
+import 'package:flutter/material.dart';
+import 'package:shop_app/models/login/login_model.dart';
+import 'package:shop_app/models/login/login_response_model.dart';
+import 'package:shop_app/modules/auth/login/bloc/login_state.dart';
+import 'package:shop_app/network/local/preference_utils.dart';
+import 'package:shop_app/network/remote/dio_helper.dart';
+import 'package:shop_app/network/remote/login_api_service.dart';
+
+import '../../../../constant.dart';
+import '../../../../network/end_points.dart';
+
+abstract class LoginRepository {
+  Future<LoginState> signInWithEmailAndPassword(LoginModel loginModel);
+}
+
+class LoginRepositoryImp extends LoginRepository {
+  @override
```

Comment:
how you handle errors Like wrong password, wrong email ...... etc
please make an exception handling class